hedera-node/hedera-app-spi/src/test/java/com/hedera/node/app/spi/config/internal/KeyValuePairConverterTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.config.internal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.hedera.node.app.spi.config.types.KeyValuePair;
+import com.swirlds.common.config.sources.SimpleConfigSource;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.config.api.ConfigurationBuilder;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class KeyValuePairConverterTest {
+
+    @Test
+    void testNullValue() {
+        // given
+        final KeyValuePairConverter converter = new KeyValuePairConverter();
+        final String value = null;
+
+        // then
+        assertThatThrownBy(() -> converter.convert(value)).isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void testValidValue() {
+        // given
+        final KeyValuePairConverter converter = new KeyValuePairConverter();
+        final String value = "key;value";
+
+        // when
+        final var result = converter.convert(value);
+
+        // then
+        assertThat(result.key()).isEqualTo("key");
+        assertThat(result.value()).isEqualTo("value");
+    }
+
+    @Test
+    void testValidValueMultipleSemicolon() {
+        // given
+        final KeyValuePairConverter converter = new KeyValuePairConverter();
+        final String value = "key;value;stillValue;";
+
+        // when
+        final var result = converter.convert(value);
+
+        // then
+        assertThat(result.key()).isEqualTo("key");
+        assertThat(result.value()).isEqualTo("value;stillValue;");
+    }
+
+    @Test
+    void testInvalidValueNoSemicolon() {
+        // given
+        final KeyValuePairConverter converter = new KeyValuePairConverter();
+        final String value = "key";
+
+        // then
+        assertThatThrownBy(() -> converter.convert(value)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testIncludedInConfig() {
+        // given
+        final Configuration configuration = ConfigurationBuilder.create()
+                .withConverter(new KeyValuePairConverter())
+                .withSource(new SimpleConfigSource("key", "key;value"))
```

Comment:
Probably never matters, but I think it is a good habit to use distinct test data for all inputs.

Reply:
I assume you problem was with the naming. I did it more concrete.